src/logic/hooks/useEstimateTransactionGas.tsx

```diff
@@ -0,0 +1,114 @@
+import { useEffect, useState } from 'react'
+import { estimateTransactionGas } from 'src/logic/safe/transactions/gas'
+import { fromTokenUnit } from 'src/logic/tokens/utils/humanReadableValue'
+import { formatAmount } from 'src/logic/tokens/utils/formatAmount'
+import { calculateGasPrice } from 'src/logic/wallets/ethTransactions'
+import { getNetworkInfo } from 'src/config'
+import { useSelector } from 'react-redux'
+import { safeParamAddressFromStateSelector, safeThresholdSelector } from 'src/logic/safe/store/selectors'
+
+export enum EstimationStatus {
+  LOADING = 'LOADING',
+  FAILURE = 'FAILURE',
+  SUCCESS = 'SUCCESS',
+}
+
+const checkIfTxIsExecution = (threshold: number, preApprovingOwner?: string, txConfirmations?: number): boolean =>
+  txConfirmations === threshold || !!preApprovingOwner || threshold === 1
```

Comment:
shouldn't we check also if the user (an owner) already signed the tx in a previous step? 
for example threshold  = 2, owners = 2
ow1 signed
ow2 signed
the hook is called and preApprovingOwner is false. If I understand correctly it will return false.

Reply:
Not sure if I understood you, in that case it should return true because txConfirmations will be 2 same as the threshold, right?

Reply:
you're right!